velox/connectors/hive/storage_adapters/hdfs/tests/InsertInfoHdfsTest.cpp

```diff
@@ -0,0 +1,152 @@
+  return std::make_shared<core::InsertTableHandle>(
+      kHiveConnectorId,
+      HiveConnectorTestBase::makeHiveInsertTableHandle(
+          outputRowType->names(),
+          outputRowType->children(),
+          partitionedBy,
+          bucketProperty,
+          HiveConnectorTestBase::makeLocationHandle(
+              outputDirectoryPath, std::nullopt, outputTableType),
+          FileFormat::PARQUET));
+}
+
+PlanNodePtr createInsertPlan(
+    PlanBuilder& inputPlan,
+    const RowTypePtr& outputRowType,
+    const std::string& outputDirectoryPath,
+    const std::vector<std::string>& partitionedBy = {},
+    const std::shared_ptr<HiveBucketProperty>& bucketProperty = {},
+    const connector::hive::LocationHandle::TableType& outputTableType =
+        connector::hive::LocationHandle::TableType::kNew,
+    const CommitStrategy& outputCommitStrategy = CommitStrategy::kNoCommit) {
+  auto insertPlan = inputPlan.tableWrite(
+      inputPlan.planNode()->outputType(),
+      outputRowType->names(),
+      nullptr,
+      createInsertTableHandle(
+          outputRowType,
+          outputTableType,
+          outputDirectoryPath,
+          partitionedBy,
+          bucketProperty),
+      bucketProperty != nullptr,
+      outputCommitStrategy);
+  return insertPlan.planNode();
+}
+
+TEST_F(InsertInfoHdfsTest, insertInfoHdfsTest) {
+  setDataTypes(
+      ROW({"c0", "c1", "c2", "c3", "c4", "c5"},
+          {BIGINT(), INTEGER(), SMALLINT(), REAL(), DOUBLE(), VARCHAR()}));
+
+  auto input = makeVectors(rowType_, 10, 100);
+
+  auto outputDirectory = "hdfs://localhost:7878/";
+  // INSERT INTO hdfs
+  auto plan = createInsertPlan(
+      PlanBuilder().values({input}), rowType_, outputDirectory);
+
+  auto result =
+      AssertQueryBuilder(plan, duckDbQueryRunner_).copyResults(pool());
+
+  auto fragmentVector = result->childAt(TableWriteTraits::kFragmentChannel)
+                            ->asFlatVector<StringView>();
+
+  const int64_t expectedRows = 10 * 100;
+  std::vector<std::string> writeFiles;
+  int64_t numRows{0};
+  int64_t fileSize{0};
+  for (int i = 0; i < result->size(); ++i) {
+    if (!fragmentVector->isNullAt(i)) {
+      ASSERT_FALSE(fragmentVector->isNullAt(i));
+      folly::dynamic obj = folly::parseJson(fragmentVector->valueAt(i));
+      ASSERT_EQ(obj["targetPath"], outputDirectory);
+      ASSERT_EQ(obj["writePath"], outputDirectory);
+      numRows += obj["rowCount"].asInt();
+
+      folly::dynamic writerInfoObj = obj["fileWriteInfos"][0];
+      const std::string writeFileName =
+          writerInfoObj["writeFileName"].asString();
+      const std::string writeFileFullPath =
+          obj["writePath"].asString() + "/" + writeFileName;
+      writeFiles.push_back(writeFileFullPath);
+      fileSize += writerInfoObj["fileSize"].asInt();
+    }
+  }
+
+  ASSERT_EQ(numRows, expectedRows);
+  ASSERT_EQ(writeFiles.size(), 1);
+
+  // Verify that the data is written to hdfs
```

Comment:
Can we use the Parquet reader to verify that the data has been written correctly?

Reply:
HI @majetideepak I added the logic to verify the content of the Parquet file, please help me review it again, thank you.